TravelAgency-spring-mvc/src/main/java/cz/muni/fi/travelAgency/controllers/CustomerController.java

```diff
@@ -45,4 +62,109 @@ public String registration(Model model) {
         return "customer/registration";
     }
 
+    @RequestMapping(value = "/create", method = RequestMethod.POST)
+    public String submitCreate(@Valid @ModelAttribute("customerCreate") CustomerCreateDTO createDTO,
+            BindingResult bindingResult, Model model, RedirectAttributes redirectAttributes,
+            UriComponentsBuilder uriBuilder) {
+        logger.debug("create(customerCreate={})", createDTO);
+        if (!resultIsValid(bindingResult, model)) {
+            return "customer/registration";
+        }
+        CustomerDTO customerDTO = new CustomerDTO(
+                createDTO.getName(),
+                createDTO.getSurname(),
+                createDTO.getEmail(),
+                createDTO.getPassword(),
+                createDTO.getPhoneNumber(),
+                createDTO.getPassportNumber(),
+                createDTO.getDateOfBirth());
+        customerFacade.registerCustomer(customerDTO, createDTO.getPassword());
+        redirectAttributes.addFlashAttribute("alert_success", "Customer " + createDTO.getId() + " was created");
+        return "redirect:" + uriBuilder.path("customer/login" + createDTO.getId()).build().encode().toUriString();
+    }
+
+    @RequestMapping(value = "/edit/{id}", method = RequestMethod.GET)
```

Comment:
the id must be retrieved from the authenticated user's information not from the adress - remove /{id}